Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -249,28 +249,64 @@ WarpXParticleContainer::DepositCurrent(WarpXParIter& pti,
 
     // Number of guard cells for local deposition of J
     WarpX& warpx = WarpX::GetInstance();
-    const int ng_J = warpx.get_ng_depos_J().max();
-
-    // Extract deposition order (same order along all directions) and check that
-    // particles shape fits within the guard cells.
-    // NOTE: In specific situations where the staggering of J and the current
-    // deposition algorithm are not trivial, this check might be too relaxed
-    // and we might include a particle that should deposit part of its current
-    // in a neighboring box. However, this should catch particles traveling many
-    // cells away, for example with algorithms that allow for large time steps.
-    const int shape_extent = static_cast<int>(WarpX::nox / 2);
+
+    const amrex::IntVect& ng_J = warpx.get_ng_depos_J();
+
```

Comment:
With https://github.com/ECP-WarpX/WarpX/pull/1607/commits/9e7f713043a1053c9b6d9a382c1d4c1d9e1b5209 we now use the new interface of `amrex::numParticlesOutOfRange`, passing an `IntVect` of guard cells.

Since different directions are taken into account automatically within the AMReX function, there is no need to duplicate these checks along x, y and z (as I did in previous commits, even though it was wrong due to the observations that you pointed out below). Hence the code went back to be about as short as it was before and I think there is no more need to use a separate function. Do you agree?

Moreover, I added a comment clarifying that for the current density J we can perform the check only on Jx, since all components of J in the different directions are allocated initially with the same numbers of guard cells.

Reply:
Yes, I agree, no need for a separate function now. :+1: